text.go

```diff
@@ -475,15 +477,18 @@ func (t *Text) logInsert(q0 int, r []rune) {
 // updated appropriately.
 func (t *Text) Insert(q0 int, r []rune, tofile bool) {
 	if !tofile {
-		panic("text.insert")
+		panic("editor.insert")
 	}
-	if tofile && t.file.HasUncommitedChanges() {
-		panic("text.insert")
+	if tofile && t.oeb.f.HasUncommitedChanges() {
+		panic("editor.insert")
 	}
 	if len(r) == 0 {
 		return
 	}
-	t.file.InsertAt(q0, r)
+	t.oeb.AllObservers(func(i interface{}) {
```

Comment:
`ObservableEditableBuffer` needs to call the observers. Wrap the functions in `File` that originally called the observers. Exposing the need to invoke the observers to the caller belies the *Observable* part of the API contract.

This way, almost no changes to the rest of Edwood should be required when you replace `File` with `undo.Buffer`  -- the changes would largely consist of modifying the wrapper functions in `ObservableEditableBuffer`.